src/main/java/Parser.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Processes user input
+ */
+public class Parser {
+
+    /** A list of Task objects to keep track of task changes*/
+    private TaskList taskList;
+
+    /**
+     * Creates a Parser object that is able to process user input
+     *
+     * @param taskList A list of task objects to keep track of task changes
+     */
+    public Parser (TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Recognises command prompts and does the appropriate task handling
+     *
+     * @param input A command prompt by the user to Duke
+     */
+    public void parse (String input) {
+        if (input.equals ("list")) {
+            taskList.list();
+        } else if (input.contains ("done")) {
+            try {
+                int taskNum = Integer.parseInt (input.substring(5));
+                taskList.done(taskNum);
+            } catch (Exception e) {
```

Comment:
Very broad Exception. You may want to create a DukeException class to help you handle the various exception in Duke!